components/http_client.rst

```diff
@@ -303,6 +348,29 @@ response sequentially instead of waiting for the entire response::
         fwrite($fileHandler, $chunk->getContent());
     }
 
+Canceling Responses
+~~~~~~~~~~~~~~~~~~~
+
+To abort a request (e.g. because it didn't complete in due time, or you want to
+fetch only the first bytes of the response, etc.), you can either:
+
+* use the ``cancel()`` method of ``ResponseInterface``::
+
+      $response->cancel()
+
+* throw an exception from a progress callback::
+
+      $response = $client->request('GET', 'https://..;', [
+        'on_progress' => function (int $dlNow, int $dlSize, array $info): void {
+            // ...
+
+            throw new \MyException();
+        },
+      ]);
+
+  The exception will be wrapped in an instance of ``TransportExceptionInterface``
```

Comment:
Unfortunately the symfony cloud endpoint for this PR is not bootstrapped yet :(
This belongs to the `* throw an exception from a progress callback::`-Part, but I am not sure if it will be rendered correct with this indention

cc @javiereguiluz

Reply:
Yes, while merging we can reword this to remove the `<ul>` list and turn it into paragraphs:

```
To abort a request (e.g. because it didn't complete in due time, or you want to
fetch only the first bytes of the response, etc.), you can either use the ``cancel()`` 
method of ``ResponseInterface``:

   ...

Or throw an exception from a progress callback::

   ...

The exception will be wrapped in an instance of ``TransportExceptionInterface``
and will abort the request.
```